### PR TITLE
Adding many stair types.

### PR DESCRIPTION
**Pull request by tristanmorgan**
Adding Stair types...
(Mossy) cobblestone_stairs,
(Smooth) quartz stairs
and (Mossy) stone brick stairs.

### CODE REVIEW

File: server/block/cobblestone_stairs.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+// SideClosed ...
+func (s CobblestoneStairs) SideClosed(pos, side cube.Pos, w *world.World) bool {
+	return s.Model().FaceSolid(pos, pos.Face(side), w)
+}
+
+// allCobblestoneStairs ...
+func allCobblestoneStairs() (stairs []world.Block) {
+	for direction := cube.Direction(0); direction <= 3; direction++ {
+		stairs = append(stairs, CobblestoneStairs{Facing: direction, UpsideDown: true, Mossy: true})
```
Comment (Prim69) [inline server/block/cobblestone_stairs.go:81]: I think it would be best here to do stairs = []world.Block{ ... }

Comment (Prim69) [reply server/block/cobblestone_stairs.go:81]: nvm i see, you can pass multiple elements to append tho

Comment (tristanmorgan) [reply server/block/cobblestone_stairs.go:81]: Thanks for the feedback, on suggestion via Discord was using reflection to DRY up the code. unfortunately I don't know how to do that yet.

File: server/block/stone_brick_stairs.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// StoneBrickStairs are blocks that allow entities to walk up blocks without jumping. They are crafted using end bricks.
+type StoneBrickStairs struct {
```
Comment (Sandertv) [inline server/block/stone_brick_stairs.go:12]: Stone brick stairs can also be `mossy`. Consider adding either a `Mossy` bool, or, similar to `StoneBricks`, a `Type StoneBricksType` field. The identifier for mossy stone bricks is `minecraft:mossy_stone_brick_stairs`.

**Review by Sandertv — COMMENTED** [review]
Got some minor comments. Mostly looks good!

File: server/block/stone_brick_stairs.go (hunk 2 above)
Comment (tristanmorgan) [reply server/block/stone_brick_stairs.go:12]: I've added the mossy variant using a boolean.

**Review by Sandertv — APPROVED** [review]
This looks good to me! Thanks a lot for the PR!

**Pull request merged**